neuraxle/hyperparams/scipy_distributions.py

```diff
@@ -0,0 +1,1010 @@
+
+    @scipy_method
+    def moment(self, n, *args, **kwargs):
+        """
+        n-th order non-central moment of distribution.
+
+        :param n:
+        :return:
+        """
+        return self.scipy_distribution.moment(n, *args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def nnlf(self, theta, x):
+        """
+        Return negative loglikelihood function.
+
+        :param theta:
+        :param x:
+        :return:
+        """
+        return self.scipy_distribution.nnlf(theta, x)
+
+    @scipy_method
+    def ppf(self, q, *args, **kwargs):
+        """
+        Percent point function(inverse of cdf) at q of the given RV.
+
+        :param q:
+        :return:
+        """
+        return self.scipy_distribution.ppf(q, *args, **kwargs)
+
+    @scipy_method
+    def sf(self, x, *args, **kwargs):
+        """
+        Survival function(1 - cdf) at x of the given RV.
+
+        :param x:
+        :return:
+        """
+        return self.scipy_distribution.sf(x, *args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def stats(self, *args, **kwargs):
+        """
+        Some statistics of the given RV.
+
+        :return:
+        """
+        return self.scipy_distribution.stats(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def std(self, *args, **kwargs):
+        """
+        Standard deviation of the distribution.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        return self.scipy_distribution.std(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def support(self, *args, **kwargs):
+        """
+        Return the support of the distribution.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        return self.scipy_distribution.support(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def var(self, *args, **kwargs):
+        """
+        Variance of the distribution.
+
+        :return:
+        """
+        return self.scipy_distribution.var(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    def to_sk_learn(self):
+        return self.scipy_distribution
+
+
+class BaseDiscreteDistribution(ScipyDistributionWrapper):
+    def __init__(self, name, min_included, max_included, null_default_value, **kwargs):
+        scipy_dist_obj = rv_discrete(
+            name=name,
+            a=min_included,
+            b=max_included,
+            **kwargs
+        )
+        scipy_dist_obj._pmf = self._pmf
+        super().__init__(
+            scipy_distribution=scipy_dist_obj,
+            null_default_value=null_default_value,
+            is_continuous=False
+        )
+
+    def _override_scipy_methods(self):
+        self.scipy_distribution._pmf = self._pmf
+
+    @abstractmethod
+    def _pmf(self, x, *args):
+        pass
+
+
+class Distribution(rv_continuous):
```

Comment:
Please rename this class. Not specific enough right now. 

Also I cant find where whis class is used because the name cant be CTRL+eFfed.